onnxruntime/python/tools/quantization/calibrate.py

```diff
@@ -23,6 +23,12 @@ def __subclasshook__(cls, subclass):
     def get_next(self) -> dict:
         """generate the input data dict for ONNXinferenceSession run"""
         raise NotImplementedError
+    
+    @property
+    @abc.abstractmethod
+    def datasize(self) -> int:
```

Comment:
there is no need to add this property. Could you please change:
            dict(zip(node_output_names, intermediate_outputs[i])) for i in range(self.data_reader.datasize) 
to:
            dict(zip(node_output_names, intermediate_output)) for intermediate_output in intermediate_outputs

Reply:
Oh, OK, didn't realize this easier solution.